src/AutoMapper/MappingEngine.cs

```diff
@@ -212,28 +190,22 @@ public TDestination DynamicMap<TDestination>(object source)
 
         public object DynamicMap(object source, Type sourceType, Type destinationType)
         {
+            ConfigurationProvider.CreateMissingTypeMaps = true;

```

Comment:
I made all the DynamicMap calls obsolete. You should just set that one config property instead of doing Map or DynamicMap. Basically the CreateMissingTypeMaps configuration was added, and it made the DynamicMap calls redundant. So prefer the config over other stuff.